### ISSUE TITLE
Unable to add leading if showMenuIcon is true.

### ISSUE THREAD

**Issue by saklanmazozgur** [egortabula/material3_layout#1]
Actually, as I mentioned in the question title, why is there such a restriction?

**Comment by egortabula** [comment]
Hello! The NavigationRail widget in Flutter only allows for one leading parameter to be set, which means that you can either display the menu icon or your own custom leading widget. However, if you have a suggestion, please feel free to share it.

**Comment by saklanmaz** [comment]
Yes, I noticed. NavigationRail does not support both. I wonder if a cheat can be done, should I try? Can the NavigationRailLabelType be set to none as soon as the NavigationRail menu button is pressed? Can it stay as all or selected again during shutdown?

**Comment by egortabula** [comment]
for NavigationRailLabelType I will try to do it today. and for leading, maybe as leading we can add a Column and as children put the showMenuIcon and some custom widget
